Memory Support: read and display memory uage

danielbeardsley opened the pull request:
Support retrieving and using the memory information that XDebug records.

Adding "memory" as cost format option was the easiest way to get it done.

This also adds the php 8.2 fixes in #169  (thanks @mmcev106)

Closes #51

Note: This went through a lot of CR and QA in our fork: https://github.com/iFixit/webgrind/pull/1

danielbeardsley committed:
Preprocessor.php: break up some long lines

Easier to read / maintian this way.

danielbeardsley committed:
PHP 8 fixes (no undefined properties).

danielbeardsley committed:
Support Memory in cachegrind files.

Xdebug has supported memory for a while, but this tool hadn't been
updated to use it.

Here we add a new "costFormat" option of "bytes of memory".
We pass this through to the pre-processor to have it read the memory
cost field instead of the time cost field.

danielbeardsley committed:
Preprocessor.cpp: read memory usage if asked

Just like the PHP pre-processor, we read the memory if costFormat ==
'bytes'.

danielbeardsley committed:
blockUI: handle jQuery upgrade

Some other pull upgraded jQuery and that caused this jQuery plugin to
crash. Stubbing this "browser" object will allow expressions like
"$.browser.ie" to return undefined instead of throwing.

danielbeardsley committed:
Table sorting: be explicit with column types

Previously, the tablesorter was detecting numeric values with thousands
separators as "text" and sorting them incorrectly.

I added thousands separators for bytes values, so they are easier to
read at a glance as the numbers get long but the sorter hadn't been
parsing them well.

Note: I'm not sure why the "Cost" column was set to "sorter: false" on
the subtables, it had still been sorting but it had been falling back to
auto-detecting the sort type.

jarstelfox committed:
Merge pull request #1 from iFixit/memory-or-time

Add Memory support

sanderdlm commented:
Can we get this merged @alpha0010 ? :)

danielbeardsley commented:
Wow... I had forgotten I did this. Thanks for commenting and reminding me.

alpha0010 commented:
Unfortunately hard to find time when my work is on other projects, but I will try to handle this soon.

danielbeardsley committed:
jQuery: fix bug in table sorter

We kept geting exceptions at this line that would break the UI
becuase h[...] was returning undefined.

danielbeardsley committed:
Commit updates to the default settings

danielbeardsley committed:
Change file list format

The 'invoked url' was the same long global path to 'dev-dispatch.php' for every
recording, so that wasn't helpful. The filename now includes the url, so the date
, filename, and size are enough.

danielbeardsley commented:
Superseded by #191 which requests to merge a feature branch instead of our master.
